### PR TITLE
Fix up khan-linter to be python3-only.

### PR DESCRIPTION
**Pull request by csilvers**
## Summary:
Now that we're on python3 everywhere, we don't need `six` anymore.  Or the `vendored/py2` directory.

I also changed all our shebang lines to say `python3` explicitly,
rather than `python`.  This isn't really needed, but some folks don't
even have a /usr/bin/python on their machine, only /usr/bin/python3!
And we might as well be clear.

While testing this, I discovered a few lint errors that are new to new
versions of python, that I cleaned up.  I also discovered that the
vendored pyflakes we have is too old for modern pythons, so I updated
it, and also mccabe while I was at it.

Issue: https://khanacademy.slack.com/archives/C02NMB1R5/p1710546934835669

## Test plan:
./runlint_test.py

### CODE REVIEW

File: hook_lib.py [hunk 1]
```diff
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
```
Comment (MiguelCastillo) [inline hook_lib.py:1]: good catch!  Presumably some refactor removed main and forgot to clean this up.  Did you need to chmod?  Im assuming it had the binary bit given it had a shebang.

File: runlint.py [hunk 2]
```diff
@@ -158,7 +155,7 @@ def _extended_fnmatch_compile(pattern):
 
 # If the code below this line has horrible syntax highlighting, check
 # this out:  http://stackoverflow.com/questions/13210816/sublime-texts-syntax-highlighting-of-regexes-in-python-leaks-into-surrounding-c
-_METACHAR_RE = re.compile(r'[[*?!]')
+_METACHAR_RE = re.compile(r'[*?![]')
```
Comment (MiguelCastillo) [inline runlint.py:158]: hard to tell which way is more readable.

File: runlint.py [hunk 3]
```diff
@@ -822,7 +819,7 @@ def main(files_and_directories,
         except Exception:
             _get_logger().error(u"ERROR linting %r with %s:\n%s" % (
                           files, type(lint_processor),
-                          traceback.format_exc().decode('utf-8')))
+                          traceback.format_exc()))
```
Comment (MiguelCastillo) [inline runlint.py:822]: I cant keep track of what needs `.decode('utf-8')` anymore. 😅

**Review by MiguelCastillo — APPROVED** [review]
Seems very reasonable to me.  Thanks for doing this cleanup!

**Comment by MiguelCastillo** [comment]
@csilvers let me know if Im blocking you from getting this one out the door :)

**Comment by csilvers** [comment]
> @csilvers let me know if Im blocking you from getting this one out the door :)

Nope!  I'm just using it locally for a while to make sure there are no hidden problems before I land it.

**Pull request merged**